tests/unit/s2n_server_new_session_ticket_test.c

```diff
@@ -72,6 +72,22 @@ static int s2n_setup_test_keys(struct s2n_connection *conn, struct s2n_config *c
     return S2N_SUCCESS;
 }
 
+bool s2n_test_session_ticket_callback_flag = false;
+static int s2n_test_session_ticket_callback(struct s2n_connection *conn,
+                                         uint8_t *session_id_data, size_t session_id_len,
+                                         uint8_t *session_data, size_t session_len,
+                                         uint32_t session_lifetime)
+{
+    EXPECT_EQUAL(session_id_len, 0);
+    EXPECT_NULL(session_id_data);
```

Comment:
These asserts are not always true right? For TLS1.3 the session ID could contain the new session ticket. I think the better way to test this would be to use the[ getter API](https://github.com/aws/s2n-tls/blob/f0872e812658a35ef5649da60a3ccf903d2c031b/api/s2n.h#L373-L384) and comparing them as mentioned below.